projects/knora-ui/src/lib/viewer/values/link-value/link-value.component.html

```diff
@@ -0,0 +1,14 @@
+<span [formGroup]="form">
+  <mat-form-field class="example-full-width linkValue-field">
+        <input [formControlName]="'linkValue'" type="text" placeholder="Link value" aria-label="resource" matInput [matAutocomplete]="auto" [readonly]="mode === 'read'">
+
+        <mat-autocomplete #auto="matAutocomplete">
+          <mat-option *ngFor="let option of options" [value]="option" (click) = "onOptionChange(option)">
```

Comment:
I think if the `FormControl` contains a `ReadResource`, then `(click) = "onOptionChange(option)"` becomes obsolete. In `getNewValue` and `getUpdatedValue` you can then directly access the `FormControl`'s value and get the IRI of the `ReadResource`.

Reply:
done!